src/internal/user_api/mod.rs

```diff
@@ -236,6 +237,115 @@ pub fn user_create<CR: rand::CryptoRng + rand::RngCore>(
+    }
+
+    /// The updated encrypted user private key
+    pub fn user_master_private_key(&self) -> &EncryptedPrivateKey {
+        &self.user_master_private_key
+    }
+
+    /// True if this user's master key requires rotation
+    pub fn needs_rotation(&self) -> bool {
+        self.needs_rotation
+    }
+}
+
+/// Get metadata about the current user
+pub fn user_get_current(
+    auth: &RequestAuth,
+) -> impl Future<Item = UserResult, Error = IronOxideErr> + '_ {
+    requests::user_get::get_curr_user(auth).and_then(|result| {
+        Ok(UserResult {
+            needs_rotation: result.needs_rotation,
+            user_public_key: result.user_master_public_key.try_into()?,
+            segment_id: result.segment_id,
+            account_id: UserId::unsafe_from_string(result.id),
+        })
+    })
+}
+
+/// Rotate the user's private key. The public key for the user remains unchanged.
+pub fn user_rotate_private_key<'apicall, CR: rand::CryptoRng + rand::RngCore>(
+    recrypt: &'apicall Recrypt<Sha256, Ed25519, RandomBytes<CR>>,
+    password: Password,
+    auth: &'apicall RequestAuth,
+) -> impl Future<Item = UserUpdatePrivateKeyResult, Error = IronOxideErr> + 'apicall {
+    requests::user_get::get_curr_user(&auth)
+        .and_then(move |curr_user| {
+            Ok({
+                let encrypt_priv_key = curr_user.user_private_key;
+                let priv_key: PrivateKey = aes::decrypt_user_master_key(
+                    &password.0,
+                    &aes::EncryptedMasterKey::new_from_slice(&encrypt_priv_key.0)?,
+                )?
+                .into();
+
+                let (new_priv_key, aug_factor) =
+                    generate_private_key_and_augmentation_factor(recrypt, &priv_key)?;
+                let new_encrypted_priv_key = aes::encrypt_user_master_key(
+                    &Mutex::new(EntropyRng::default()),
+                    &password.0,
+                    new_priv_key.as_bytes(),
+                )?;
+                (
+                    UserId(curr_user.id),
+                    curr_user.current_key_id,
+                    new_encrypted_priv_key,
+                    aug_factor,
+                )
+            })
+        })
+        .and_then(
+            move |(user_id, curr_key_id, new_encrypted_priv_key, aug_factor)| {
+                requests::user_update_private_key::update_private_key(
+                    &auth,
+                    user_id,
+                    curr_key_id,
+                    new_encrypted_priv_key.into(),
+                    aug_factor.into(),
+                )
+                .map(|resp| resp.into())
+            },
+        )
+}
+
+/// Generate a new private key and augmentation factor from a given PrivateKey.
+/// There is a very small chance that an augmentation factor could not be compatible with
+/// the given PrivateKey, so we retry once internally before giving the caller an error.
+fn generate_private_key_and_augmentation_factor<R: KeyGenOps>(
```

Comment:
I agree that this name isn't very good. I've moved this code to internal/mod.rs as it will be needed for group private key rotation (I think) and renamed it as `augment_private_key_with_retry`